internal/guidedremediation/manifest/maven/pomxml.go

```diff
@@ -63,3 +73,313 @@ type DependencyWithOrigin struct {
 	maven.Dependency
 	Origin string // Origin indicates where the dependency comes from
 }
+
+type mavenManifest struct {
+	filePath     string
+	root         resolve.Version
+	requirements []resolve.RequirementVersion
+	groups       map[manifest.RequirementKey][]string
+	specific     ManifestSpecific
+}
+
+// FilePath returns the path to the manifest file.
+func (m *mavenManifest) FilePath() string {
+	return m.filePath
+}
+
+// Root returns the Version representing this package.
+func (m *mavenManifest) Root() resolve.Version {
+	return m.root
+}
+
+// System returns the ecosystem of this manifest.
+func (m *mavenManifest) System() resolve.System {
+	return resolve.Maven
+}
+
+// Requirements returns all direct requirements (including dev).
+func (m *mavenManifest) Requirements() []resolve.RequirementVersion {
+	return m.requirements
+}
+
+// Groups returns the dependency groups that the direct requirements belong to.
+func (m *mavenManifest) Groups() map[manifest.RequirementKey][]string {
+	return m.groups
+}
+
+// LocalManifests returns Manifests of any local packages.
+func (m *mavenManifest) LocalManifests() []manifest.Manifest {
+	return nil
+}
+
+// EcosystemSpecific returns any ecosystem-specific information for this manifest.
+func (m *mavenManifest) EcosystemSpecific() any {
+	return m.specific
+}
+
+// Clone returns a copy of this manifest that is safe to modify.
+func (m *mavenManifest) Clone() manifest.Manifest {
+	clone := &mavenManifest{
+		filePath:     m.filePath,
+		root:         m.root,
+		requirements: slices.Clone(m.requirements),
+		groups:       maps.Clone(m.groups),
+		specific: ManifestSpecific{
+			Parent:                 m.specific.Parent,
+			Properties:             slices.Clone(m.specific.Properties),
+			OriginalRequirements:   slices.Clone(m.specific.OriginalRequirements),
+			RequirementsForUpdates: slices.Clone(m.specific.RequirementsForUpdates),
+			Repositories:           slices.Clone(m.specific.Repositories),
+		},
+	}
+	clone.root.AttrSet = m.root.AttrSet.Clone()
+
+	return clone
+}
+
+type readWriter struct {
+	*datasource.MavenRegistryAPIClient
+}
+
+// GetReadWriter returns a ReadWriter for pom.xml manifest files.
+func GetReadWriter(registry string) (manifest.ReadWriter, error) {
+	client, err := datasource.NewMavenRegistryAPIClient(datasource.MavenRegistry{URL: registry, ReleasesEnabled: true})
+	if err != nil {
+		return nil, err
+	}
+	return readWriter{MavenRegistryAPIClient: client}, nil
+}
+
+// System returns the ecosystem of this ReadWriter.
+func (r readWriter) System() resolve.System {
+	return resolve.Maven
+}
+
+// Read parses the manifest from the given file.
+func (r readWriter) Read(path string, fsys scalibrfs.FS) (manifest.Manifest, error) {
```

Comment:
There's a decent amount of duplicated logic from the `pomxmlnet` extractor (that also existed in osv-scanner). 
I'm not sure if we want to extract some common functionality out of them both?